client/src/game/ui/notes/TagAutoCompleteSearch.vue

```diff
@@ -0,0 +1,227 @@
+<script setup lang="ts" generic="T extends { name: string, colour: string }">
+import { computed, ref, watch } from "vue";
+
+import { mostReadable } from "../../../core/utils";
+
+const emit = defineEmits<{
+    (e: "picked", payload: T): void;
+    (e: "update:modelValue", payload: string): void;
+}>();
+
+const props = defineProps<{
+    options: T[];
+    placeholder: string;
+    modelValue?: string;
+}>();
+
+const searchQuery = ref(props.modelValue ?? "");
+
+const searchBar = ref<HTMLInputElement | null>(null);
+
+watch(() => props.modelValue, (newValue) => {
+    if (newValue !== undefined) {
+        searchQuery.value = newValue;
+    }
+});
+
+// https://stackoverflow.com/a/69901745
```

Comment:
Is this comment still relevant? I don't really see a `as Ref<` anywhere?

Reply:
I forget why exactly I added that comment now. Your'e right it's no longer relevant, I just forgot to remove it when I changed results to `computed`